gazebo/common/Dem_TEST.cc

```diff
@@ -180,8 +180,19 @@ TEST_F(DemTest, NegDem)
   // Check the heights and widths
   EXPECT_EQ(33, static_cast<int>(dem.GetHeight()));
   EXPECT_EQ(33, static_cast<int>(dem.GetWidth()));
-  EXPECT_FLOAT_EQ(293.51068, dem.GetWorldHeight());
-  EXPECT_FLOAT_EQ(293.51089, dem.GetWorldWidth());
+  // This DEM model is from the moon. Older versions
+  // of libproj will calculate the size assuming it
+  // is of the Earth, unless we specify the surface.
+  bool sizeSameAsEarth =
+    (std::abs(293.51089 - dem.GetWorldWidth()) < 0.1)
+    && (std::abs(293.51068 - dem.GetWorldHeight()) < 0.1);
+  // Newer versions give invalid sizes, 0 in this case.
+  bool invalidSize =
+    (dem.GetWorldHeight() < 0.001) &&
+    (dem.GetWorldWidth() < 0.001);
+
+  EXPECT_TRUE(sizeSameAsEarth || invalidSize);
```

Comment:
Is it possible to use the `libproj` macros? https://github.com/OSGeo/PROJ/blob/35e4207a9d9eb86b4e5188f00eae1f152306dad3/src/proj.h#L175-L177

or 

https://github.com/OSGeo/PROJ/blob/35e4207a9d9eb86b4e5188f00eae1f152306dad3/src/proj.h#L188

eg. 
```c++
#if PROJ_AT_LEAST_VERSION(<ubuntu 18.04 version>)
  EXPECT_FLOAT_EQ(293.51068, dem.GetWorldHeight());
  EXPECT_FLOAT_EQ(293.51089, dem.GetWorldWidth());
#else
  ...
#endif
```

Reply:
I did try that initially (https://github.com/gazebosim/gazebo-classic/pull/3275/commits/82b92ca96c80212bea940cc6ea8bc8226b9fbcda), but the CI was complaining that it couldn't find "proj.h". Worked for me locally though.

Reply:
I believe that will be resolved if you add `find_package(PROJ REQUIRED)` then add `${PROJ_INCLUDE_DIRS}` to `include_directories`, and `${PROJ_LIBRARIES}` to `target_link_libraries`

https://github.com/OSGeo/PROJ/blob/65f60f6e27482920aa5007af5357d1f37b220b90/cmake/CMakeLists.txt#L14-L21

eg., 

Update:
https://github.com/gazebosim/gazebo-classic/blob/09fb1bca67d96b34a124e9a8fe355d7ffc9de6fa/gazebo/common/CMakeLists.txt#L16-L18

To be:
```Cmake
if (HAVE_GDAL)
  include_directories(${GDAL_INCLUDE_DIR})
  if (PROJ_FOUND)
    include_directories(${PROJ_INCLUDE_DIRS})
  endif()
endif()
```

I'm not sure what version of `PROJ` is being used but it may need `PROJ4` instead: https://github.com/OSGeo/PROJ/blob/65f60f6e27482920aa5007af5357d1f37b220b90/cmake/CMakeLists.txt#L51-L56